### PR TITLE
Fix volume code to consider existing pvc

### PR DESCRIPTION
**Pull request by gnufied**
Fixes `oc set volume` such that, if user is adding a new mounted volume from same PVC, rather than creating new `pod.Spec.Volumes` - we reuse existing one.

Partly fixes - https://bugzilla.redhat.com/show_bug.cgi?id=1527685

### CODE REVIEW

**Comment by gnufied** [comment]
cc @childsb

**Comment by gnufied** [comment]
@soltysh added tests

File: pkg/oc/cli/cmd/set/volume.go [hunk 1]
```diff
@@ -699,41 +699,60 @@ func (v *VolumeOptions) getVolumeName(spec *kapi.PodSpec, singleResource bool) (
...
+	if len(v.Output) == 0 {
+		fmt.Fprintf(v.Err, "info: Generated volume name: %s\n", name)
+	}
+	return name, false, nil
+}
+
+func (v *VolumeOptions) checkForExistingClaim(spec *kapi.PodSpec) (string, bool) {
+	for _, vol := range spec.Volumes {
+		oldSource := vol.VolumeSource.PersistentVolumeClaim
+		if oldSource != nil && v.AddOpts.ClaimName == oldSource.ClaimName {
```
Comment (rootfs) [inline pkg/oc/cli/cmd/set/volume.go]: no namespace?

Comment (gnufied) [reply pkg/oc/cli/cmd/set/volume.go]: There is no need of checking for namespace here. Both PVC and Pod will belong to same namespace.

**Pull request merged**